test/integration-tests/gas-tests/NarwhalVSGenie.gas.test.ts

```diff
@@ -0,0 +1,168 @@
+import {
+  seaportOrders,
+  seaportInterface,
+  getAdvancedOrderParams,
+  AdvancedOrder,
+} from '../shared/protocolHelpers/seaport'
+import { COVEN_721, ENS_721, GENIE_SWAP, resetFork } from '../shared/mainnetForkHelpers'
+import { ALICE_ADDRESS, DEADLINE, OPENSEA_CONDUIT_KEY } from '../shared/constants'
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers'
+import hre from 'hardhat'
+import deployUniversalRouter, { deployPermit2 } from '../shared/deployUniversalRouter'
+import { Contract, BigNumber } from 'ethers'
+import { genieX2Y2MarketInterface, TradeDetails } from '../shared/protocolHelpers/genieSwap'
+import { x2y2Orders, X2Y2_INTERFACE } from '../shared/protocolHelpers/x2y2'
+import { expect } from 'chai'
+const { ethers } = hre
+import fs from 'fs'
+
+describe.only('NFT UX Tests gas', () => {
+  let alice: SignerWithAddress
+  let router: UniversalRouter
+  let permit2: Permit2
+  let planner: RoutePlanner
+  let genieSwap: Contract
+
+  beforeEach(async () => {
+    await resetFork()
+    await hre.network.provider.request({
+      method: 'hardhat_impersonateAccount',
+      params: [ALICE_ADDRESS],
+    })
+    alice = await ethers.getSigner(ALICE_ADDRESS)
+    permit2 = (await deployPermit2()).connect(alice) as Permit2
+    router = (await deployUniversalRouter(permit2)).connect(alice) as UniversalRouter
+    planner = new RoutePlanner()
+    genieSwap = GENIE_SWAP.connect(alice)
+  })
+
+  describe('Seaport', () => {
+    let seaportCalldata: string
+    let value: BigNumber
+    let advancedOrder: AdvancedOrder
+    let cryptoCovens: ERC721
+
+    beforeEach(async () => {
+      ;({ advancedOrder, value } = getAdvancedOrderParams(seaportOrders[0]))
+      seaportCalldata = seaportInterface.encodeFunctionData('fulfillAdvancedOrder', [
+        advancedOrder,
+        [],
+        OPENSEA_CONDUIT_KEY,
+        alice.address,
+      ])
+      cryptoCovens = COVEN_721.connect(alice) as ERC721
+    })
+
+    describe('Narwhal', async () => {
+      it('ETH -> NFT', async () => {
+        planner.addCommand(CommandType.SEAPORT, [value.toString(), seaportCalldata])
+        const { commands, inputs } = planner
+
+        await snapshotGasCost(router['execute(bytes,bytes[],uint256)'](commands, inputs, DEADLINE, { value }))
+        const ownerAfter = await cryptoCovens.ownerOf(advancedOrder.parameters.offer[0].identifierOrCriteria)
+        expect(ownerAfter.toLowerCase()).to.eq(ALICE_ADDRESS)
+      })
+    })
+
+    describe('Genie', async () => {
+      it('ETH -> NFT', async () => {
+        let tradeDetails: TradeDetails[] = [
+          {
+            marketId: 21,
+            value: value,
+            tradeData: seaportCalldata,
+          },
+        ]
+
+        await snapshotGasCost(
+          genieSwap.multiAssetSwap(
+            [[], []],
+            [],
+            [],
+            [],
+            tradeDetails,
+            ['0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE'],
```

Comment:
This could be a constant.

```suggestion
            [E_BURN_ADDRESS],
```